### PR TITLE
fix: improve scaling and textures

### PR DESCRIPTION
**Pull request by ooystein**
- Implement scaling for completion layer
- Improve cement and screen textures
- Create hole texture based on max diameter of holes

### CODE REVIEW

File: src/layers/CompletionLayer.ts [hunk 1]
```diff
@@ -146,26 +138,26 @@ export class CompletionLayer<T extends Completion[]> extends WellboreBaseCompone
...
+      canvas.width = size;
+      canvas.height = size;
       const canvasCtx = canvas.getContext('2d');
 
       if (canvasCtx instanceof CanvasRenderingContext2D) {
         canvasCtx.fillStyle = 'white';
         canvasCtx.fillRect(0, 0, canvas.width, canvas.height);
 
-        const baseLineWidth = 6;
+        const baseLineWidth = size / 10; // eslint-disable-line no-magic-numbers
```
Comment (ovineq) [inline src/layers/CompletionLayer.ts:155]: Same here, consider adding "10" to the list of exceptions.

File: src/layers/CustomDisplayObjects/FixedWidthSimpleRope.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  /**
+   * Note: The wrap mode of the texture is set to REPEAT if `textureScale` is positive.
+   * @param texture - The texture to use on the rope.
+   * @param points - An array of {@link PIXI.Point} objects to construct this rope.
+   * @param {number} textureScale - Optional. Positive values scale rope texture
+   * keeping its aspect ratio. You can reduce alpha channel artifacts by providing a larger texture
+   * and downsampling here. If set to zero, texture will be stretched instead.
+   * @param width - Width of rope
+   */
+  constructor(texture: Texture, points: IPoint[], textureScale = 0, width: number) {
```
Comment (ovineq) [inline src/layers/CustomDisplayObjects/FixedWidthSimpleRope.ts:23]: Can you reorder `width` and `textureScale`, as the latter has a default.

File: src/layers/CustomDisplayObjects/FixedWidthSimpleRope.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const ropeGeometry = new RopeGeometry(width, points, textureScale);
+    const meshMaterial = new MeshMaterial(texture);
+
+    if (textureScale > 0) {
+      // attempt to set UV wrapping, will fail on non-power of two textures
+      texture.baseTexture.wrapMode = WRAP_MODES.REPEAT;
+    }
```
Comment (ovineq) [inline src/layers/CustomDisplayObjects/FixedWidthSimpleRope.ts:30]: Should we include this comment in the contructor docs aswell?

**Review by ovineq — APPROVED** [review]
Great work! 🥳  
I only have suggestions below:

File: src/layers/CasingAndCementLayer.ts [hunk 4]
```diff
@@ -222,23 +224,23 @@ export class CasingAndCementLayer<T extends CasingAndCementData> extends Wellbor
...
-
-    canvasCtx.lineWidth = 1;
+    canvasCtx.lineWidth = lineWidth;
     canvasCtx.fillStyle = secondCementColor;
-
     canvasCtx.beginPath();
-    canvasCtx.lineWidth = 1;
 
-    const distanceBetweenLines = 10;
+    const distanceBetweenLines = size / 12; // eslint-disable-line no-magic-numbers
```
[the remark this answers is not in the document]
Comment (ooystein) [reply src/layers/CasingAndCementLayer.ts:243]: then I'd rather create a const for it. But I'm not to sure it makes things more readable. 
const distanceBetweenLines = size / proportionalDistanceBetweenLines

File: src/layers/CustomDisplayObjects/FixedWidthSimpleRope.ts (hunk 3 above)
Comment (ooystein) [reply src/layers/CustomDisplayObjects/FixedWidthSimpleRope.ts:30]: It's nothing special with this spesific rope. Its a common thing with using textures in WebGL

File: src/layers/CasingAndCementLayer.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (ovineq) [reply src/layers/CasingAndCementLayer.ts:243]: As long as we're adding the exceptions on line-bases, it's fine.
Resolving the other one as well, as it's the same case.

File: src/layers/CustomDisplayObjects/FixedWidthSimpleRope.ts (hunk 3 above)
Comment (ooystein) [reply src/layers/CustomDisplayObjects/FixedWidthSimpleRope.ts:30]: fixed it

**Pull request merged**